### ISSUE TITLE
rclone v1.52.2, Swift, Segments: overwrite segmenting file - do break this file

### ISSUE THREAD

**Issue by Artemeey** [rclone/rclone#4469]
version: 
rclone v1.52.2
- os/arch: linux/amd64
- go version: go1.14.4

storage system: Swift

please create random test.txt file with size 2M

```
echo 1 >> test.txt
rclone --swift-chunk-size 1M copy test.txt remote:/tmp
```

this ok

```
echo 1 >> test.txt
rclone --swift-chunk-size 1M copy test.txt remote:/tmp

rclone lsl selectel:backup/test.txt
# result: 0 2020-07-29 09:20:35.095788368 test.txt
```

this not ok
sometimes: old segments doesnot remove
sometimes: new segments not associate with file test.txt

manual fix

```
echo 1 >> test.txt

rclone delete remote:/tmp/test.txt
rclone delete remote:/tmp_segments/test.txt
rclone --swift-chunk-size 1M copy test.txt remote:/tmp
```

this is ok

debug data upload:
```
rclone -vv copy --swift-chunk-size 1M test.txt selectel:backup 
2020/07/29 09:20:24 DEBUG : rclone: Version "v1.52.2" starting with parameters ["rclone" "-vv" "copy" "--swift-chunk-size" "1M" "test.txt" "selectel:backup"] 
2020/07/29 09:20:24 DEBUG : Using config file from "/root/.config/rclone/rclone.conf" 
2020/07/29 09:20:24 DEBUG : fs cache: renaming cache item "test.txt" to be canonical "/dataHDD/temp/backuper" 
2020/07/29 09:20:24 DEBUG : test.txt: Need to transfer - File not found at Destination 
2020/07/29 09:20:24 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000000" into "backup_segments" 
2020/07/29 09:20:24 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000001" into "backup_segments" 
2020/07/29 09:20:25 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000002" into "backup_segments" 
2020/07/29 09:20:25 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000003" into "backup_segments" 
2020/07/29 09:20:25 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000004" into "backup_segments" 
2020/07/29 09:20:25 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000005" into "backup_segments" 
2020/07/29 09:20:25 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000006" into "backup_segments" 
2020/07/29 09:20:25 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000007" into "backup_segments" 
2020/07/29 09:20:25 DEBUG : test.txt: Uploading segment file "test.txt/1596003624.935929806/8460163/00000008" into "backup_segments" 
2020/07/29 09:20:25 DEBUG : test.txt: Uploading segments into "backup_segments" seems done (EOF) 
2020/07/29 09:20:25 DEBUG : test.txt: Returning empty Md5sum for swift large object 
2020/07/29 09:20:25 INFO : test.txt: Copied (new) 
2020/07/29 09:20:25 INFO : 
Transferred: 8.068M / 8.068 MBytes, 100%, 7.848 MBytes/s, ETA 0s 
Transferred: 1 / 1, 100% 
Elapsed time: 1.0s
```

debug data overwrite:
```
root@db-spb:/dataHDD/temp/backuper# rclone -vv copy --swift-chunk-size 1M test.txt selectel:backup                                                                                                                                                                      
2020/07/29 09:20:38 DEBUG : rclone: Version "v1.52.2" starting with parameters ["rclone" "-vv" "copy" "--swift-chunk-size" "1M" "test.txt" "selectel:backup"]                                                                                                           
2020/07/29 09:20:38 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"                                                                                                                                                                                   
2020/07/29 09:20:38 DEBUG : fs cache: renaming cache item "test.txt" to be canonical "/dataHDD/temp/backuper"                                                                                                                                                           
2020/07/29 09:20:38 DEBUG : test.txt: Sizes differ (src 8460165 vs dst 8460163)                                                                                                                                                                                         
2020/07/29 09:20:38 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000000" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:38 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000001" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:38 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000002" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:38 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000003" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:38 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000004" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:38 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000005" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:39 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000006" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:39 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000007" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:39 DEBUG : test.txt: Uploading segment file "test.txt/1596003638.64122954/8460165/00000008" into "backup_segments"                                                                                                                                     
2020/07/29 09:20:39 DEBUG : test.txt: Uploading segments into "backup_segments" seems done (EOF)                                                                                                                                                                        
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000000" in container "backup_segments"                                                                                                                              
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000001" in container "backup_segments"                                                                                                                              
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000002" in container "backup_segments"                                                                                                                              
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000003" in container "backup_segments"                                                                                                                              
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000004" in container "backup_segments"                                                                                                                              
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000005" in container "backup_segments"                                                                                                                              
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000006" in container "backup_segments"                                                                                                                              
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000007" in container "backup_segments"                                                                                                                              
2020/07/29 09:20:39 DEBUG : test.txt: Removing segment file "test.txt/1596003638.64122954/8460165/00000008" in container "backup_segments"
2020/07/29 09:20:39 DEBUG : test.txt: Returning empty Md5sum for swift large object
2020/07/29 09:20:39 INFO  : test.txt: Copied (replaced existing)
2020/07/29 09:20:39 INFO  : 
Transferred:        8.068M / 8.068 MBytes, 100%, 7.100 MBytes/s, ETA 0s
Transferred:            1 / 1, 100%
Elapsed time:         1.1s
```

See:
```
...
Uploading segment file "test.txt/1596003638.64122954/8460165/00000000
...
Removing segment file "test.txt/1596003638.64122954/8460165/00000000
...
```

Is normal?
Real name old segment is not 1596003638.64122954
Dir 1596003638.64122954 not created

**Comment by ncw** [comment]
Both your logs looked OK.

When it goes wrong are there errors in the log?

**Comment by Artemeey** [comment]
> Both your logs looked OK.
> 
> When it goes wrong are there errors in the log?

Uploading segment file "test.txt/1596003638.64122954/8460165/00000000
Removing segment file "test.txt/1596003638.64122954/8460165/00000000

Uploading and Removing - 1596003638.64122954, this name is incorrect
Name directory for removing must be - 1596003624.935929806, but not 1596003638.64122954

**Comment by ncw** [comment]
I have replicated the problem

```
$ truncate -s 2M test.txt
$ 
$ rclone  --swift-chunk-size 1M copy test.txt memstore:test-issue-4469
$ rclone lsl memstore:test-issue-4469
  2097152 2020-08-02 12:34:43.137540349 test.txt
$ 
$ echo 1 >> test.txt
$ rclone  --swift-chunk-size 1M copy test.txt memstore:test-issue-4469
$ rclone lsl memstore:test-issue-4469
        0 2020-08-02 12:35:34.897894519 test.txt
$
```

In more depth

```
$ rclone  --swift-chunk-size 1M copy test.txt memstore:test-issue-4469
$ rclone lsf -R memstore:test-issue-4469_segments
test.txt/1596368258.239491886/2097154/00000000
test.txt/1596368258.239491886/2097154/00000001
test.txt/1596368258.239491886/2097154/00000002
test.txt/
test.txt/1596368258.239491886/
test.txt/1596368258.239491886/2097154/
$ 
$ rclone lsl memstore:test-issue-4469
  2097154 2020-08-02 12:35:34.897894519 test.txt
$ 
$ echo 1 >> test.txt
$ 
$ rclone  -vv --swift-chunk-size 1M copy test.txt memstore:test-issue-4469
2020/08/02 12:38:12 DEBUG : rclone: Version "v1.52.2-018-g029f817e-v1.52-stable-beta" starting with parameters ["rclone" "-vv" "--swift-chunk-size" "1M" "copy" "test.txt" "memstore:test-issue-4469"]
2020/08/02 12:38:12 DEBUG : Using config file from "/home/ncw/.rclone.conf"
2020/08/02 12:38:12 DEBUG : fs cache: adding new entry for parent of "test.txt", "/tmp"
2020/08/02 12:38:12 DEBUG : test.txt: Sizes differ (src 2097156 vs dst 2097154)
2020/08/02 12:38:12 DEBUG : test.txt: Uploading segment file "test.txt/1596368292.598008394/2097156/00000000" into "test-issue-4469_segments"
2020/08/02 12:38:13 DEBUG : test.txt: Uploading segment file "test.txt/1596368292.598008394/2097156/00000001" into "test-issue-4469_segments"
2020/08/02 12:38:14 DEBUG : test.txt: Uploading segment file "test.txt/1596368292.598008394/2097156/00000002" into "test-issue-4469_segments"
2020/08/02 12:38:15 DEBUG : test.txt: Uploading segments into "test-issue-4469_segments" seems done (EOF)
2020/08/02 12:38:15 DEBUG : test.txt: Removing segment file "test.txt/1596368292.598008394/2097156/00000000" in container "test-issue-4469_segments"
2020/08/02 12:38:15 DEBUG : test.txt: Removing segment file "test.txt/1596368292.598008394/2097156/00000001" in container "test-issue-4469_segments"
2020/08/02 12:38:16 DEBUG : test.txt: Removing segment file "test.txt/1596368292.598008394/2097156/00000002" in container "test-issue-4469_segments"
2020/08/02 12:38:16 DEBUG : test.txt: Returning empty Md5sum for swift large object
2020/08/02 12:38:16 INFO  : test.txt: Copied (replaced existing)
2020/08/02 12:38:16 INFO  : 
Transferred:   	    2.000M / 2.000 MBytes, 100%, 465.564 kBytes/s, ETA 0s
Transferred:            1 / 1, 100%
Elapsed time:         4.3s

2020/08/02 12:38:16 DEBUG : 4 go routines active
$ rclone lsl memstore:test-issue-4469
        0 2020-08-02 12:38:01.942900841 test.txt
$ rclone lsf -R memstore:test-issue-4469_segments
test.txt/1596368258.239491886/2097154/00000000
test.txt/1596368258.239491886/2097154/00000001
test.txt/1596368258.239491886/2097154/00000002
test.txt/
test.txt/1596368258.239491886/
test.txt/1596368258.239491886/2097154/
$ 
```

The problem comes here

```
2020/08/02 12:39:47 DEBUG : HTTP REQUEST (req 0xc000193b00)
2020/08/02 12:39:47 DEBUG : HEAD /v1/AUTH_XXX/test-issue-4469/test.txt HTTP/1.1
Host: csproxy1.memset.com
User-Agent: rclone/v1.52.2-018-g029f817e-v1.52-stable-beta
Transfer-Encoding: chunked
X-Auth-Token: XXXX

0

2020/08/02 12:39:47 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/08/02 12:39:47 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/08/02 12:39:47 DEBUG : HTTP RESPONSE (req 0xc000193b00)
2020/08/02 12:39:47 DEBUG : HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Sun, 02 Aug 2020 11:39:47 GMT
Etag: "d41d8cd98f00b204e9800998ecf8427e"
Last-Modified: Sun, 02 Aug 2020 11:38:15 GMT
X-Object-Manifest: test%2Dissue%2D4469%5Fsegments/test.txt/1596368292.598008394/2097156
X-Object-Meta-Mtime: 1596368281.942900841
X-Timestamp: 1596368295.33033
X-Trans-Id: tx1fbb014649b842e3971bc-005f26a603
Content-Length: 0
```

Where we can see that the manifest has `test%2Dissue%2D4469%5Fsegments/test.txt/1596368292.598008394/2097156`

Whereas the data is in `test.txt/1596368258.239491886/2097154/00000000`

So it looks like rclone failed to update the manifest...

**Comment by ncw** [comment]
I see what has happened, rclone is uploading the segments then immediately deleting them again.

This was due to an old code re-organisation.

Try this

[v1.52.2-281-g324077fb-fix-4075-swift-delete-beta](https://beta.rclone.org/branch/v1.52.2-281-g324077fb-fix-4075-swift-delete-beta/) on branch [fix-4075-swift-delete](https://github.com/rclone/rclone/tree/fix-4075-swift-delete) (uploaded in 15-30 mins)

**Comment by Artemeey** [comment]
> I see what has happened, rclone is uploading the segments then immediately deleting them again.
> 
> This was due to an old code re-organisation.
> 
> Try this
> 
> [v1.52.2-281-g324077fb-fix-4075-swift-delete-beta](https://beta.rclone.org/branch/v1.52.2-281-g324077fb-fix-4075-swift-delete-beta/) on branch [fix-4075-swift-delete](https://github.com/rclone/rclone/tree/fix-4075-swift-delete) (uploaded in 15-30 mins)

This works, thank you!

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.53

Thanks for testing :-)
